pkg/gui/controllers/commit_message_controller.go

```diff
@@ -56,8 +85,67 @@ func (self *CommitMessageController) context() types.Context {
 	return self.contexts.CommitMessage
 }
 
+func (self *CommitMessageController) handlePreviousCommit() error {
+	return self.handleCommitIndexChange(1)
+}
+
+func (self *CommitMessageController) handleNextCommit() error {
+	if self.commitMessageContext.GetSelectedIndex() == 0 {
+		return nil
+	}
+	return self.handleCommitIndexChange(-1)
+}
+
+func (self *CommitMessageController) handleCommitDescriptionPress() error {
+	if err := self.c.PushContext(self.contexts.CommitDescription); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (self *CommitMessageController) handleCommitIndexChange(value int) error {
+	self.commitMessageContext.IncrementSelectedIndexBy(value)
+	currentIndex := self.commitMessageContext.GetSelectedIndex()
+	if currentIndex == 0 {
+		self.setCommitMessage("")
+		return nil
+	}
+	return self.setCommitMessageAtIndex(currentIndex)
+}
+
+func splitMessageAndDescription(commitMessage string) (string, string) {
+	// TODO: when saving the commit + description with the CommitCmdObj it creates two \n between the message & description
+	parts := strings.Split(commitMessage, "\n\n")
+	msg := parts[0]
+	var description string
+	if len(parts) > 1 {
+		description = parts[1]
+	}
+	return msg, description
+}
+
+func (self *CommitMessageController) setCommitMessageAtIndex(index int) error {
+	msg, err := self.git.Commit.GetCommitMessageFromHistory(index)
```

Comment:
What happens if we move the index to before the first commit in the repo? We should ensure that doesn't cause any problems (we could just do nothing in the event of that situation and continue showing the current message)

Reply:
I think if the index you search for is > the index of the first commit it always returns the first commit, so we would show that same message. But the counter would keep increasing if someone spams up arrow. Not sure if we can control that because how would we know when to stop incrementing?

Reply:
> I think

We need to be sure :)

Crashing with `index out of bounds` is something we would like to avoid if possible. 

If what you say is true, that it's capped, since each commit has a unique hash (let's pretend it is) hash, we'll know we've hit rock bottom because two neighbouring commits have the same hash.